pytorch_lightning/trainer/trainer.py

```diff
@@ -467,27 +475,49 @@ def fit(
         self.accelerator_backend.setup(self, model)
         self.setup_trainer(model)
 
+        # ----------------------------
+        # INSPECT THE CORE LOOPS
+        # ----------------------------
+        # Our fitting flow looks like this.
+        #
+        #   trainer.test(...) or trainer.predict(...)   ||
+        #                     |                         ||
+        #               trainer.fit(...)                ||
```

Comment:
why are test and predict before fit?

Reply:
Because `.test` and `.predict` will call `.fit`.